src/react-count-to.js

```diff
@@ -13,48 +12,59 @@ const CountTo = React.createClass({
     className: React.PropTypes.string,
   },
 
+  getDefaultProps() {
+    return {
+      from: 0,
+      delay: 100,
+      digits: 0,
+      onComplete: () => {},
+    };
+  },
+
   getInitialState() {
     return {
-      counter: this.props.from || 0,
+      counter: this.props.from,
     };
   },
 
   componentDidMount() {
-    this.start(this.props);
+    this.start();
   },
 
   componentWillReceiveProps(nextProps) {
-    if (!shallowequal(this.props, nextProps)) {
-      this.start(nextProps);
+    const props = this.props;
+
+    // no need to restart animation if onComplete / digits / className changes
+    if (nextProps.to !== props.to || nextProps.from !== props.from
+    || nextProps.speed !== props.speed || nextProps.delay !== props.delay) {
+      this.start();
     }
   },
 
   componentWillUnmount() {
     this.clear();
   },
 
-  start(props) {
+  start() {
     this.clear();
     this.setState(this.getInitialState(), () => {
-      const delay = this.props.delay || 100;
+      const delay = this.props.delay;
       this.loopsCounter = 0;
-      this.loops = Math.ceil(props.speed / delay);
-      this.increment = (props.to - this.state.counter) / this.loops;
-      this.interval = setInterval(this.next.bind(this, props), delay);
+      this.loops = Math.ceil(this.props.speed / delay);
+      this.increment = (this.props.to - this.state.counter) / this.loops;
+      this.interval = setInterval(this.next, delay);
     });
   },
 
-  next(props) {
+  next() {
     if (this.loopsCounter < this.loops) {
       this.loopsCounter++;
       this.setState({
         counter: this.state.counter + this.increment,
       });
     } else {
       this.clear();
-      if (props.onComplete) {
-        props.onComplete();
-      }
+      this.props.onComplete();
```

Comment:
We should check again here for the callback to exists, since it has no default value (see my comment above):

```javascript
if (this.props.onComplete) {
  this.props.onComplete();
}
```